canonical-web-and-design/maas-ui
# feat(ui): build device summary page

## Done

- Built device summary page. The design is a bit different to legacy to make it look more like the machine summary.
- Made the machine summary network card reusable across node types and improved responsiveness

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /MAAS/r/devices and click on a device
- Check that you are taken to the summary tab and all the data is correct
- Check that the page looks alright on all screen sizes

## Fixes

Fixes canonical-web-and-design/app-tribe#536

## Screenshot
### Large screen
![Screenshot 2021-12-03 at 16-49-47 test summary bolla MAAS](https://user-images.githubusercontent.com/25733845/144558606-7d22a30f-8237-455c-92f8-e1611c3587e3.png)

### Small screen
![Screenshot 2021-12-03 at 16-56-07 test summary bolla MAAS](https://user-images.githubusercontent.com/25733845/144558971-7f9e02f4-22bf-4d83-a7b9-7a9ba5981a53.png)
